app/views/administrate/application/edit.html.erb

```diff
@@ -27,7 +27,7 @@ It displays a header, and renders the `_form` partial to do the heavy lifting.
       "#{t("administrate.actions.show")} #{page.page_title}",
       [namespace, page.resource],
       class: "button",
-    ) if valid_action? :show %>
+    ) if valid_action?(:show) && show_action?(:show, page.resource) %>
```

Comment:
Is it possible to use only `show_action?` in all cases?
Like
```ruby
def show_action?(action, resource)
  valid_action?(action, resource) && authorized?(action, resource)
end
```

Reply:
Hi @pustomytnyk - `#show_action?` is a method that users of administrate can (and should) override in their controllers. If we implement it in terms of `#valid_action?` in the base controller and again in the Pundit mixin, there's still opportunity for administrate users to accidentally break the expected behaviour by overriding it to something that's not.